pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -295,19 +291,14 @@ func (exp *flowExporter) sendFlowRecords() error {
 				}
 				exp.numDataSetsSent = exp.numDataSetsSent + 1
 			}
-			// If the connection is not present in the conntrack table anymore,
-			// then we delete the connection in connection store and record store.
-			// Please note that our record cache policy is that we keep the record in
-			// the record map as long as the corresponding connection is present in
-			// the conntrack table.
-			if !record.Conn.IsPresent {
-				klog.V(2).Infof("Deleting the inactive connection with key: %v from both connection map and record map", key)
-				if err := exp.connStore.DeleteConnectionByKey(key); err != nil {
-					return err
-				}
+			// If connection is in dying state, we will delete all the flow records.
+			if flowexporter.IsConnectionDying(&record.Conn) {
+				klog.V(2).Infof("Deleting the inactive connection with key: %v from record map", key)
 				if err := exp.flowRecords.DeleteFlowRecordWithoutLock(key); err != nil {
 					return err
 				}
+				conn, _ := exp.connStore.GetConnByKey(key)
+				conn.IsExportedAndDying = true
```

Comment:
TBH, I'd rather separate concerns here and have an `IsExported` flag (set by the exporter) and an `IsDying` flag (if needed, set based on the connection state)

Reply:
We found that extra flag is not necessary in our case. So removed them. Thanks